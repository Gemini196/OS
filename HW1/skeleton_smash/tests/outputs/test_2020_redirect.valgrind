==44800== Memcheck, a memory error detector
==44800== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==44800== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==44800== Command: /home/student/Desktop/OS/HW1/skeleton_smash/smash
==44800== Parent PID: 44798
==44800== 
==44800== 
==44800== FILE DESCRIPTORS: 6 open at exit.
==44800== Open file descriptor 1: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_redirect.out
==44800==    at 0x54EF957: dup2 (syscall-template.S:78)
==44800==    by 0x112CC9: main (smash.cpp:41)
==44800== 
==44800== Open file descriptor 0:
==44800==    at 0x54EF957: dup2 (syscall-template.S:78)
==44800==    by 0x112C77: main (smash.cpp:33)
==44800== 
==44800== Open file descriptor 5:
==44800==    at 0x54EF927: dup (syscall-template.S:78)
==44800==    by 0x112E17: __static_initialization_and_destruction_0(int, int) (smash.cpp:9)
==44800==    by 0x112E33: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==44800==    by 0x112EAC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==44800==    by 0x5400C17: (below main) (libc-start.c:266)
==44800== 
==44800== Open file descriptor 4: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_redirect.out
==44800==    at 0x54EF927: dup (syscall-template.S:78)
==44800==    by 0x112E07: __static_initialization_and_destruction_0(int, int) (smash.cpp:8)
==44800==    by 0x112E33: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==44800==    by 0x112EAC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==44800==    by 0x5400C17: (below main) (libc-start.c:266)
==44800== 
==44800== Open file descriptor 3: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_redirect.valgrind
==44800==    <inherited from parent>
==44800== 
==44800== Open file descriptor 2: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_redirect.err
==44800==    <inherited from parent>
==44800== 
==44800== 
==44800== HEAP SUMMARY:
==44800==     in use at exit: 0 bytes in 0 blocks
==44800==   total heap usage: 280 allocs, 280 frees, 99,945 bytes allocated
==44800== 
==44800== All heap blocks were freed -- no leaks are possible
==44800== 
==44800== For counts of detected and suppressed errors, rerun with: -v
==44800== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
