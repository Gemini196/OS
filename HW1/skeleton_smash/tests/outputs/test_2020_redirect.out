smash> smash> smash> smash> smash> test_redirect
smash> smash> again_redirect
smash> smash> again_redirect
test_redirect
smash> smash> smash pid is 49779
smash> smash> smash pid is 49779
smash pid is 49779
smash> smash> smash pid is 49779
smash> mypid
new
smash> smash> smash> [1] sleep 100& : 49825 0 secs
smash> smash> smash> smash> smash> k
mypid
new
s
t
smash> smash> k
mypid
new
p
s
t
smash> k
mypid
new
p
s
t
smash> smash> smash> before
smash> again_redirect
test_redirect
smash> smash> smash> smash> k
mypid
p
s
t
smash> smash> k
mypid
new
p
s
t
smash> k
mypid
new
p
s
t
smash> smash> smash> k
mypid
new
p
s
t
smash> k
mypid
new
p
s
t
smash> smash> smash> k
mypid
new
p
s
t
smash> smash> smash> k
mypid
new
p
s
t
test
smash> smash> smash: got ctrl-Z
smash: process 49779 was stopped
smash: got ctrl-Z
